db/compaction_picker.cc

```diff
@@ -682,6 +682,10 @@ Compaction* CompactionPicker::CompactRange(
     return nullptr;
   }
 
+  if (inputs.empty() && output_level_inputs.empty()) {
```

Comment:
There shouldn't be any way to reach this point with empty `inputs`. If there is, let's fix it, but I couldn't find any way. I wrote some questions about your use case in the issue.